deno.json

```diff
@@ -1,7 +1,18 @@
 {
   "name": "@logtape/logtape",
   "version": "0.5.0",
-  "exports": "./logtape/mod.ts",
+  "exports": {
+    ".": "./logtape/mod.ts",
+    "./config": "./logtape/config.ts",
+    "./filesink": "./logtape/filesink.jsr.ts",
+    "./filter": "./logtape/filter.ts",
+    "./formatter": "./logtape/formatter.ts",
+    "./fs": "./logtape/fs.ts",
+    "./level": "./logtape/level.ts",
+    "./logger": "./logtape/logger.ts",
+    "./record": "./logtape/record.ts",
+    "./sink": "./logtape/sink.ts"
+  },
```

Comment:
The current `exports` are very carefully curated by me. The APIs that are not `export`ed are not exposed for a reason. Are there any APIs that you would like to see exposed?

Reply:
For some background, I'm basically trying to wrap the logtape loggers in a abstract logger class, the goal is basically to be able to register the various loggers that someone wants to use, and to be able to use them as such,

```ts
// Interface to extend Logger with custom loggers
export type LoggerWithCustomLoggers<L extends string> = Logger<L> & {
  [P in L]: Logger<L>;
}

// Function to create a logger with typed custom loggers
export function createLogger<L extends string>(options: LoggerOptions<L> = {}): LoggerWithCustomLoggers<L> {
  return new Logger(options) as LoggerWithCustomLoggers<L>;
}

const logger = createLogger({
  loggers: {
    init: initLogger,
    build: buildLogger,
    customLogger: getLogger(["@bundle/core", "custom"]),
  },
});

// Logs to the general logger by default
logger.info`General Log ${5}`;

// Logs to the init logger with a custom template
logger.with({ logger: "init", message: "start {time}" }).warn({ time: Date.now() });

// Logs an error to the init logger
logger.with({ logger: "init" }).error("error {message}", { message: "Error message" });

// Logs an error to the init logger (with context)
logger.with({ logger: "init" }).context({ message: "Error message" }).error("error {message}");

// Logs an error to the init logger (with context no. 2)
logger.with({ logger: "init", context: { message: "Error message" } }).error("error {message}");

// Shortcut for init logger
logger.init.error("Initialization failed: {error}", { error: new Error("cool") });

// Logs to the custom logger
logger.customLogger.info("Custom logger in action");

// Shortcut for build logger
logger.build.fatal`A fatally cool message`;

// This will now cause a TypeScript error, as 'nonExistentLogger' is not registered
// logger.nonExistentLogger.info("This should not compile");
```

To achieve this workflow I need to add context to the logger, but that get's complicated as the original logtape Logger class doesn't really support the concept of contextual logs (though it is possible to replicate). 

In addition, logtape has a couple of unhandled edge cases that context can help with, e.g. what happens if the user wants to log out a variable that ends up being a function, etc... For this to work I need to be able to fully control how log records are emitted, that's the main reason for needing all the modules to be exported in the exports field.

Reply:
The API's I would need access to are basically all types and interfaces, the `LoggerImpl` class,  the `log`, `logLazily`, `logTemplate` & `emit` methods, and `parseMessageTemplate` & `renderMessage` functions.

Reply:
Looking at your gist you linked to, I still don't understand how `LoggerImpl` and its methods are used. 🤔

Basically, I don't want to expose things like `LoggerImpl` because the wider the API surface, the harder it is to change the internal implementation without breaking backwards compatibility.

However, if you have a feature that you need in LogTape, it may be better to modify LogTape itself rather than extend it externally. I'm not sure what exactly the *context* you're trying to implement is for though. 😅

Reply:
For `LoggerImpl` if possible exporting it as a type would potentially solve that issue. But I would still need access to the `parseMessageTemplate` & `renderMessage` functions. Not sure what can be done about that.

Reply:
I've made a couple changes to the pr, I've created a new `types.ts` module where I just export all the types instead of exporting all the code, I then export `parseMessageTemplate` and `renderMessage` in the `mod.ts` file, what do you think?